rust/src/storage/s3/mod.rs

```diff
@@ -163,14 +165,53 @@ impl S3StorageBackend {
             Region::default()
         };
 
-        let client = create_s3_client(region).unwrap();
-        Self { client }
+        let client = create_s3_client(region.clone())?;
+        let lock_client = try_create_lock_client(region)?;
+
+        Ok(Self {
+            client,
+            lock_client,
+        })
+    }
+
+    async fn unsafe_rename_obj(
```

Comment:
IMHO - I think we should log a `warn!` anytime this method is called.

Reply:
I reuse it in both locked/non-locked envs, so i feel like we should find something else here, or even change function name to be less misleading

Reply:
Ah true. Maybe a warn in https://github.com/delta-io/delta-rs/pull/212/files#diff-87e43fff59e072c5dbd99295f15fbf86d5a0e53f6586689191b2c61e07ace91aR417 on startup without a locking provider would be better.